utils/wasm-gen/src/tests.rs

```diff
@@ -157,6 +167,245 @@ fn injecting_addresses_works() {
+
+            let mut payload = Payload::default();
+            payload.extend_with(0);
+
+            params_config.add_rule(
+                ParamType::Size,
+                // offset = length = MAX_PAYLOAD_SIZE / 2, so MAX_PAYLOAD_SIZE / 2 would be read
+                // when the payload with max possible length is provided.
+                SysCallParamAllowedValues::constant(payload.inner().len() as i64 / 2),
+            );
+
+            (params_config, None, payload)
+        }
+        SysCallName::Read => {
+            let mut params_config = SysCallsParamsConfig::constant_value(0);
+            params_config.add_rule(
+                ParamType::Size,
+                SysCallParamAllowedValues::constant(i32::MAX as i64),
+            );
+            (params_config, None, Payload::default())
+        }
+        SysCallName::PayProgramRent => (
+            SysCallsParamsConfig::constant_value(0),
+            Some(MemoryWrite {
+                offset: 0,
+                content: std::iter::repeat(255).take(128).collect(),
+            }),
+            Payload::default(),
+        ),
+        _ => (
+            SysCallsParamsConfig::constant_value(0),
+            None,
+            Payload::default(),
+        ),
+    }
+}
+
+fn execute_wasm_with_syscall_injected(
+    syscall: SysCallName,
+    ignore_fallible_errors: bool,
+    params_config: SysCallsParamsConfig,
+    initial_memory_write: Option<MemoryWrite>,
+    payload: Payload,
+) -> TerminationReason {
+    use gear_backend_common::{BackendReport, Environment};
+    use gear_backend_wasmi::WasmiEnvironment;
+    use gear_core::{
+        gas::{GasAllowanceCounter, GasCounter, ValueCounter},
+        memory::AllocationsContext,
+        message::{ContextSettings, DispatchKind, IncomingDispatch, MessageContext},
+        reservation::GasReserver,
+    };
+    use gear_core_processor::{configs::PageCosts, ProcessorContext, ProcessorExternalities};
+
+    const INITIAL_PAGES: u16 = 1;
+    const INJECTED_SYSCALLS: u32 = 8;
+
+    let buf = vec![0; UNSTRUCTURED_SIZE];
+    let mut unstructured = Unstructured::new(&buf);
+
+    let mut injection_amounts = SysCallsInjectionAmounts::all_never();
+    injection_amounts.set(syscall, INJECTED_SYSCALLS, INJECTED_SYSCALLS);
+
+    let gear_config = (
+        GearWasmGeneratorConfigBuilder::new()
+            .with_memory_config(MemoryPagesConfig {
+                initial_size: INITIAL_PAGES as u32,
+                ..MemoryPagesConfig::default()
+            })
+            .with_sys_calls_config(
+                SysCallsConfigBuilder::new(injection_amounts)
+                    .with_params_config(params_config)
+                    .set_ignore_fallible_syscall_errors(ignore_fallible_errors)
+                    .build(),
+            )
+            .with_entry_points_config(EntryPointsSet::Init)
+            .with_recursions_removed(true)
```

Comment:
seems odd when only one function is generated

Reply:
but it's enough for us, as we testing the logics independent of generated function count